File: 1944-number-of-visible-people-in-a-queue/1944-number-of-visible-people-in-a-queue.java
class Solution {
    public int[] canSeePersonsCount(int[] heights) {
    Stack<Integer>s= new Stack<Integer>();
    int []arr = new int[heights.length];
        arr[heights.length-1]=0;
    s.push(heights[heights.length-1]);
    // System.out.println(s.peek());
    for(int i=heights.length-2;i>=0;--i)
    {
        if(heights[i]<heights[i+1])
        {
            arr[i]=1;
            s.push(heights[i]);
        }
        
        else
        {
            int c = 0;
    
            while(!s.empty() && heights[i]>s.peek())
            {
                // System.out.println(c+" "+s.pop());
                s.pop();
                c++;
            }
            if(!s.empty())
                arr[i]++;
            arr[i]+=c;
            s.push(heights[i]);
        }
    
    }
        return arr;
    }
}

File: 1944-number-of-visible-people-in-a-queue/1944-number-of-visible-people-in-a-queue.cpp
class Solution {
public:
    vector<int> canSeePersonsCount(vector<int>& heights) {
        stack<int>s;
        vector<int>ans(heights.size());
        s.push(heights.size()-1);
        for(int i = heights.size()-2;~i;--i)
        {
            int ct = 0;
            while(!s.empty()&&heights[s.top()]<heights[i])
                s.pop(),ct++;
            if(!s.empty())ct++;
            s.push(i);
            ans[i]=ct;
        }
        return ans;
    }
};